.gitlab-ci.yml

```diff
@@ -0,0 +1,152 @@
+variables:
+  PROJECT_ALLOC_NAME: ${CI_PROJECT_NAME}_ci_${CI_PIPELINE_ID}
+  BUILD_ROOT: ${CI_PROJECT_DIR}
+  # Should make it so we aren't doing fresh clones of GEOSX all the time
+  GIT_STRATEGY: fetch
+  # Manually update submodules
+  GIT_SUBMODULE_STRATEGY: none
+
+stages:
+  - build
+  - nightly
+
+####
+# Template
+.build_script:
+  script:
+    # Change submodule paths to absolute github paths (default is relative to GEOSX gitlab path)
+    - git submodule set-url src/coreComponents/LvArray https://github.com/GEOSX/LvArray.git
+    - git submodule set-url src/cmake/blt https://github.com/LLNL/blt.git
+    - git submodule set-url src/coreComponents/constitutive/PVTPackage https://github.com/GEOSX/PVTPackage.git
+    - git submodule set-url integratedTests git@github.com:GEOSX/integratedTests.git
+    - git submodule set-url src/coreComponents/mesh/PAMELA https://github.com/GEOSX/PAMELA.git
+    - git submodule set-url src/coreComponents/fileIO/coupling/hdf5_interface https://github.com/GEOSX/hdf5_interface.git
+
+    # Update submodules
+    - git submodule update --init --recursive src/cmake/blt
+    - git submodule update --init --recursive src/coreComponents/LvArray
+    - git submodule update --init --recursive src/coreComponents/constitutive/PVTPackage
+    - git submodule update --init --recursive integratedTests
+    - git submodule update --init src/coreComponents/mesh/PAMELA
+    - git submodule update --init --recursive src/coreComponents/fileIO/coupling/hdf5_interface
+
+    - mkdir ${SYSTEM}
```

Comment:
Is `SYSTEM` "cleaned" after each build?
Does it contain some `JOB_ID` that make it unique?

Reply:
Each job is run in its own unique directory, so there will not be conflicts between each build.